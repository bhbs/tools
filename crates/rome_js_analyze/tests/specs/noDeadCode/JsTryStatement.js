function JsTryStatement1() {
    try {
        test();
        return;
    } catch (err) {
        test();
        return;
    }

    afterTryCatchReturn();
}
