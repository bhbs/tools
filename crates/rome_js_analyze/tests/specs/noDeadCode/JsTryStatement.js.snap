---
source: crates/rome_js_analyze/tests/spec_tests.rs
expression: JsTryStatement.js
---
# Input
```js
function JsTryStatement1() {
    try {
        test();
        return;
    } catch (err) {
        test();
        return;
    }

    afterTryCatchReturn();
}

```

# Diagnostics
```
warning[noDeadCode]: This code is unreachable
   ┌─ JsTryStatement.js:10:5
   │
 4 │         return;
   │         ------- Either this statement ...
   ·
 7 │         return;
   │         ------- ... or this statement will return from the function ...
   ·
10 │     afterTryCatchReturn();
   │     ---------------------- ... before it can reach this code


```


